pandas/core/resample.py

```diff
@@ -856,10 +856,32 @@ def size(self):
         # It's a special case as higher level does return
         # a copy of 0-len objects. GH14962
         result = self._downsample("size")
-        if not len(self.ax) and isinstance(self._selected_obj, ABCDataFrame):
+        if not len(self.ax):
+            if isinstance(self._selected_obj, ABCDataFrame):
+                from pandas import Series
+
+                result = Series([], index=result.index, dtype="int64")
+            elif isinstance(self._selected_obj, ABCSeries):
+                from pandas import Series
+
+                result = Series(
+                    [], index=result.index, dtype="int64", name=self._selected_obj.name
+                )
+            else:
+                raise AssertionError("Unexpected type result")
+
+        return result
+
+    @Appender(GroupBy.count.__doc__)
+    def count(self):
+        result = self._downsample("count")
+        if not len(self.ax) and isinstance(self._selected_obj, ABCSeries):
```

Comment:
Does this need to handle DataFrame like `size`?

Reply:
Yes, it was needed as well. I added this and added the relevant test cases.